src/main/java/org/openlmis/stockmanagement/domain/LotOnHand.java

```diff
@@ -21,4 +29,21 @@
   @JsonFormat(shape=JsonFormat.Shape.STRING, pattern="yyyy-MM-dd")
   @JsonDeserialize(using=DateDeserializer.class)
   Date effectiveDate;
+
+  private LotOnHand(Lot lot, StockCard stockCard) {
+    Objects.requireNonNull(lot);
+    Objects.requireNonNull(stockCard);
+    this.lot = lot;
+    this.stockCard = stockCard;
+    this.quantityOnHand = 0L;
+    this.effectiveDate = new Date();
+  }
+
+  public void addToQuantityOnHand(long quantity) {
+    this.quantityOnHand += quantity;
+  }
+
+  public static final LotOnHand createZeroedLotOnHand(Lot lot, StockCard stockCard) {
```

Comment:
Possibly add comment/javadoc here to indicate will throw exception if lot or stockCard is null